tests/transport/can/media/_socketcand.py

```diff
@@ -0,0 +1,153 @@
+    pytest.skip("Socketcand test skipped because the system is not GNU/Linux", allow_module_level=True)
+
+
+GIBIBYTE = 1024**3
+
+MEMORY_LIMIT = 8 * GIBIBYTE
+"""
+The test suite artificially limits the amount of consumed memory in order to avoid triggering the OOM killer
+should a test go crazy and eat all memory.
+"""
+
+_logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.fixture()
+def configure_host_environment() -> None:
+    print("configurng host environment")
+
+    def execute(
+        *cmd: typing.Any, ensure_success: bool = True, cwd: typing.Optional[str] = None, daemon: bool = False
+    ) -> typing.Tuple[int, str, str]:
+        cmd = tuple(map(str, cmd))
+        if daemon:
+            subprocess.Popen(
+                cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True
+            )  # start subproccess without waiting for output
+            return 0
+
+        out = None
+        if cwd is None:
+            out = subprocess.run(  # pylint: disable=subprocess-run-check
+                cmd,
+                encoding="utf8",
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+        else:
+            out = subprocess.run(  # pylint: disable=subprocess-run-check
+                cmd, encoding="utf8", stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=cwd
+            )
+
+        stdout, stderr = out.stdout, out.stderr
+        _logger.debug("%s stdout:\n%s", cmd, stdout)
+        _logger.debug("%s stderr:\n%s", cmd, stderr)
+        if out.returncode != 0 and ensure_success:  # pragma: no cover
+            raise subprocess.CalledProcessError(out.returncode, cmd, stdout, stderr)
+        assert isinstance(stdout, str) and isinstance(stderr, str)
+        return out.returncode, stdout, stderr
+
+    if sys.platform.startswith("linux"):
+        import resource  # pylint: disable=import-error
+
+        _logger.info("Limiting process memory usage to %.1f GiB", MEMORY_LIMIT / GIBIBYTE)
+        resource.setrlimit(resource.RLIMIT_AS, (MEMORY_LIMIT, MEMORY_LIMIT))
+
+        # Set up virtual SocketCAN interfaces.
+        execute("sudo", "modprobe", "can")
+        execute("sudo", "modprobe", "can_raw")
+        execute("sudo", "modprobe", "vcan")
+        execute("sudo", "ip", "link", "add", "dev", "vcan3", "type", "vcan", ensure_success=False)
+        execute("sudo", "ip", "link", "set", "vcan3", "mtu", 72)  # Enable both Classic CAN and CAN FD.
+        execute("sudo", "ip", "link", "set", "up", "vcan3")
+
+        # build and install socketcand
+        execute("sudo", "apt-get", "install", "-y", "autoconf")
+        execute("git", "clone", "https://github.com/linux-can/socketcand.git")
+        execute("./autogen.sh", cwd="socketcand")
+        execute("./configure", cwd="socketcand")
+        execute("make", cwd="socketcand")
+        execute("sudo", "make", "install", cwd="socketcand")
```

Comment:
Move this into the GitHub workflow or noxfile

Reply:
I might be mistaken but for local testing GH workflow isn't used, and nox is. So I think it would make more sense for me to put it in the nox file so that it can be run. I also thought it may be possible to put it the conftest.py file. Do you have a preference on any of these